tools/federation/src/main/java/org/eclipse/rdf4j/federated/FedXConnection.java

```diff
@@ -388,6 +389,21 @@ private static int getOriginalMaxExecutionTime(BindingSet b) {
 		return 0;
 	}
 
+	/**
+	 * Return the original explicit includeInferred {@link Operation#getIncludeInferred()} value.
+	 * 
+	 * @param b
+	 * @return the supplied value or {@link FedXConfig#getIncludeInferredDefault()}
+	 */
+	private boolean getOriginalIncludeInferred(BindingSet b) {
```

Comment:
We really should get a possibility inside the SAIL to access information on the Query object (e.g. in this case the includeInferred), I am using nasty workarounds by passing it as a binding

Reply:
I haven't looked at this long so perhaps I'm missing something obvious, but  I don't understand why you need this at all. The `includeInferred` parameter is an input for `evaluateInternal`. Why do you need this passing arround through a config object at all?

Reply:
@jeenbroekstra thanks for this. You are absolutetly right, includeInferred is available in evaluateInternal. Did not look properly or have missed it. This will make my change a bit simpler.
I will provide a revised version later.

Still for the query execution time there is no way to access it inside the sail.

Reply:
Done, I amended the change and could get rid of this piece of code